include/aca_on_demand_engine.h

```diff
@@ -17,22 +17,49 @@
 
 #define IPTOS_PREC_INTERNETCONTROL 0xc0
 #define DSCP_DEFAULT (IPTOS_PREC_INTERNETCONTROL >> 2)
-#define STDOUT_FILENO   1   /* Standard output.  */
+#define STDOUT_FILENO 1 /* Standard output.  */
 
 #include "common.pb.h"
 #include <openvswitch/ofp-errors.h>
 #include <openvswitch/ofp-packet.h>
 #include <string>
+#include <thread>
+#include "hashmap/HashMap.h"
+#include <grpcpp/grpcpp.h>
+#include <unordered_map>
+#include "aca_log.h"
 
 using namespace alcor::schema;
-
+using namespace std;
+// using namespace grpc;
+struct on_demand_payload {
+  std::chrono::_V2::steady_clock::time_point insert_time;
+  string uuid;
+  uint32_t in_port;
+  void *packet;
+  int packet_size;
+  alcor::schema::Protocol protocol;
+};
 // ACA on-demand engine implementation class
 namespace aca_on_demand_engine
 {
 class ACA_On_Demand_Engine {
   public:
+  /* This thread is responsible for processing hostOperationReplies from NCM */
+  std::thread *on_demand_reply_processing_thread;
+  /* 
+  This thread checks request_uuid_on_demand_data_map periodically and removes any 
+  entry that has been staying in the map for more than ON_DEMAND_ENTRY_CLEANUP_FREQUENCY_IN_MICROSECONDS
+  */
+  std::thread *on_demand_payload_cleaning_thread;
+  grpc::CompletionQueue _cq;
+  unordered_map<std::string, on_demand_payload *, std::hash<std::string> > request_uuid_on_demand_data_map;
+  std::mutex _mutex;
```

Comment:
@zzxgzgz can we give it a more descriptive name for `_mutex`, since it is used by `request_uuid_on_demand_data_map` only.

Reply:
fixed